src/solver/main.cpp

```diff
@@ -112,53 +97,64 @@ const char* const ANTARES_LOGO = "\n\n"
                                  "                . : = * # % % % % # + - .                   \n"
                                  "                    . . : : : : : . .                       \n\n";
 
-static void NotEnoughMemory()
+constexpr int ALLOCATION_FAILURE_EXIT_CODE = 42;
+
+void logAbortion()
 {
-    logs.fatal() << "Not enough memory. aborting.";
-    AntaresSolverEmergencyShutdown(42);
+    if (!logs.logfile())
+    {
+        logs.fatal() << "Aborting now. (warning: no file log available)";
+        logs.warning() << "No log file available";
+    }
+    else
+    {
+        logs.error() << "Aborting now. See logs for more details";
+    }
+}
+
 }
 
 /*!
 ** \brief main
 */
 int main(int argc, char** argv)
 {
-    logs.info(ANTARES_LOGO);
-    logs.info(GPL_ANNOUNCEMENT);
-    // Name of the running application for the logger
-    logs.applicationName("solver");
-
-    // Dealing with the lack of memory
-    std::set_new_handler(&NotEnoughMemory);
+    try {
 
-    if (not memory.initializeTemporaryFolder())
-        return EXIT_FAILURE;
+        logs.info(ANTARES_LOGO);
+        logs.info(GPL_ANNOUNCEMENT);
+        // Name of the running application for the logger
+        logs.applicationName("solver");
 
-    // locale
-    InitializeDefaultLocale();
+        if (not memory.initializeTemporaryFolder())
+            return EXIT_FAILURE;
 
-    // Getting real UTF8 arguments
-    argv = AntaresGetUTF8Arguments(argc, argv);
-    Antares::Solver::Application application;
-    try
-    {
-        application.prepare(argc, argv);
-    }
-    // Catch errors
-    catch (const Error::LoadingError& e)
-    {
-        logs.error() << e.what();
-        AntaresSolverEmergencyShutdown();
-    }
+        // locale
+        InitializeDefaultLocale();
 
-    application.execute();
+        // Getting real UTF8 arguments
+        argv = AntaresGetUTF8Arguments(argc, argv);
 
-    application.writeExectutionInfo();
+        Antares::Solver::Application application;
+        application.prepare(argc, argv);
+        application.execute();
+        application.writeExectutionInfo();
 
-    FreeUTF8Arguments(argc, argv);
+        FreeUTF8Arguments(argc, argv);
 
-    // to avoid a bug from wxExecute, we should wait a little before returning
-    SuspendMilliSeconds(200 /*ms*/);
+        // to avoid a bug from wxExecute, we should wait a little before returning
+        SuspendMilliSeconds(200 /*ms*/);
```

Comment:
TODO: remove

Reply:
I left it for that PR: do we create an issue for this, to be done once GUI is actually removed ?